presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -170,15 +170,22 @@ public static ColumnMapping decimalColumnMapping(DecimalType decimalType, Roundi
         if (decimalType.isShort()) {
             return ColumnMapping.longMapping(
                     decimalType,
-                    (resultSet, columnIndex) -> encodeShortScaledValue(resultSet.getBigDecimal(columnIndex), scale),
+                    shortDecimalReadFunction(decimalType, UNNECESSARY),
                     shortDecimalWriteFunction(decimalType));
         }
         return ColumnMapping.sliceMapping(
                 decimalType,
-                (resultSet, columnIndex) -> encodeScaledValue(resultSet.getBigDecimal(columnIndex).setScale(scale, roundingMode)),
+                longDecimalReadFunction(decimalType, roundingMode),
                 longDecimalWriteFunction(decimalType));
     }
 
+    public static LongReadFunction shortDecimalReadFunction(DecimalType decimalType, RoundingMode roundingMode)
```

Comment:
Maybe there should be an overload with only the type argument and `UNNECESSARY` rounding (and for `longDecimalReadFunction`)?

Reply:
Good idea, added